static/js/pm_list.js

```diff
@@ -147,15 +225,149 @@ export function is_all_privates() {
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // If we do not find any other active pm in our convos then we clear the list of PMs and
+        // collapse the section.
+        $(".expanded-private-messages").empty();
+        close();
+    }
+}
+
+export function activate_all_private_messages_view() {
+    $("#private_messages_section").addClass("active_private_messages_section");
+    $(".more-private-messages-sidebar-title").css("font-weight", "bold");
+}
+
+export function deactivate_all_private_messages_view() {
+    $("#private_messages_section").removeClass("active_private_messages_section");
+    $(".more-private-messages-sidebar-title").css("font-weight", "normal");
+}
+
+export function handle_narrow_activated(filter) {
+    const active_filter = filter;
+    // We check whether the new narrow is also related to a PM thread or not
+    if (active_filter.operands("pm-with").length !== 0) {
+        // If we find the new narrow to also be a PM thread then check for number of PMs present and state of PM section
+        if (
+            // If we find all PMs to be collapsed and only 1 previously active PM outside of the collapsed
+            // pms, then we remove the previous one and append the newly active PM inside the collapsed PMs.
+            only_active_pm_out_of_collapsed_pms()
+        ) {
+            append_new_active_pm_in_collapsed_pms();
+        }
+        // After activating narrow for a PM thread, we deactivate the `All private messages` view.
+        deactivate_all_private_messages_view();
+    } else {
+        // If we find no PMs in narrow filter and we see that state of PM section is
+        // suppose to be collapsed we then empty out the list of PMs and close it.
+        if (private_messages_collapsed) {
+            $(".expanded-private-messages").empty();
+            close();
+        }
+        // We always deactivate the previously activated PM in case of new narrow not belonging to PMs.
+        $(".expanded-private-messages li.active-sub-filter").removeClass("active-sub-filter");
+    }
+}
+
+export function handle_narrow_deactivated() {
+    // In case of the new narrow not belonging to PMs, we revert the state of current PM section
+    // to it's initial state (even in the cae of zoomed PM section) and deactivate the previously activated
+    // PM and resize the stream bar again.
+    hide_more_pms();
+    deactivate_all_private_messages_view();
+    $("#private_messages").removeClass("zoom-in").addClass("zoom-out");
+    $("#streams_list").show();
+    $(".left-sidebar .right-sidebar-items").show();
+    if (private_messages_collapsed) {
+        $(".expanded-private-messages").empty();
+        close();
+    }
+    $(".expanded-private-messages li.active-sub-filter").removeClass("active-sub-filter");
+    update_private_messages();
+    setTimeout(() => {
+        resize.resize_stream_filters_container();
+    }, 0);
```

Comment:
This type of `setTimeout` is a code smell and often represents a bug; please document with a comment why the delay is necessary.

Reply:
I understand that this type of code is a mess and was ofc difficult for me also to debug to this solution  -- I've seen this sort of thing implemented at some other places as well, In any case I'll comment out a reason behind this.